src/EventStore.ClusterNode/Program.cs

```diff
@@ -181,5 +167,41 @@ internal static class Program {
 				Log.CloseAndFlush();
 			}
 		}
+
+		private static ServerOptionsSelectionCallback CreateServerOptionsSelectionCallback(ClusterVNodeHostedService hostedService) {
+			return ((_, _, _, _) => {
+				var serverOptions = new SslServerAuthenticationOptions {
+					ServerCertificateContext = SslStreamCertificateContext.Create(
+						hostedService.Node.CertificateSelector(),
+						hostedService.Node.IntermediateCertificatesSelector(),
+						offline: true),
+					ClientCertificateRequired = true,
```

Comment:
It seems like this statement and the subsequent comment directly contradict each other...

Reply:
@thefringeninja well spotted, i'll add a comment to clarify. the name `ClientCertificateRequired` is a bit misleading - it only means "Request a client certificate" but the client has the possibility not to supply one.

Reply:
The RFC says:

```
7.4.6.  Client Certificate

When this message will be sent:

This is the first message the client can send after receiving a ServerHelloDone message.  This message is only sent if the server requests a certificate. If no suitable certificate is available, the client MUST send a certificate message containing no certificates.
```

Reply:
Fixed in https://github.com/EventStore/EventStore/pull/3398/commits/0bd39c33e98682d8eee4da32efbd29ce821789f0